xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/attachment_macros.vm

```diff
@@ -55,94 +55,98 @@
+    #if ($fieldName.indexOf('.') < 0)
+      #set ($fieldNameWithAlias = 'attachment.' + $fieldName)
+    #end
+    #set ($matchType = $request.getParameter("${fieldName}_match"))
+    #if ("$!matchType" == '')
+      #set ($matchType = $defaultMatchType)
     #end
-    #set ($options = {
-      'cssClass': "#if ($foreach.index % 2 == 0)even#{else}odd#end",
-      'redirect': $redirect
-    })
-    #displayAttachment($attachment $options)
+    #set ($parameterName = $fieldNameWithAlias.replace('.', '_'))
+    #if ($matchType == 'exact')
+      #set ($discard = $constraints.add("$fieldNameWithAlias = :$parameterName"))
+      #set ($parameterValue = $request.getParameter($fieldName))
+    #else
+      #set ($discard = $constraints.add("lower($fieldNameWithAlias) like lower(:$parameterName)"))
+      #set ($parameterValue = $request.getParameter($fieldName).trim())
+    #end
+    #set ($discard = $queryParameters.add({
+      'name': $parameterName,
+      'match': $matchType,
+      'value': $parameterValue
+    }))
   #end
 #end
 
-#macro (displayAttachment $attachment $options)
-  ## Compute the attachment reference because there's no getter.
-  #set ($attachmentReference = $services.model.createAttachmentReference($attachment.document.documentReference,
-    $attachment.filename))
-  <div class="attachment $!escapetool.xml($options.cssClass)">
-    <span class="mime" data-type="$!escapetool.xml($attachment.mimeType)">
-      #if ($attachment.isImage())
-        <span title="$escapetool.xml($services.localization.render('core.viewers.attachments.mime.image'))">
-          <img src="$xwiki.getURL($attachmentReference, 'download', 'width=48')"
-            alt="$escapetool.xml($attachment.filename)"/>
-        </span>
-      #else
-        #mimetypeimg($attachment.mimeType.toLowerCase() $attachment.filename.toLowerCase())
-      #end
-    </span>
-    <div class="information">
-      <span class="name">
-        <a href="$xwiki.getURL($attachmentReference)" data-entity-type="$escapetool.xml($attachmentReference.type)"
-          data-entity-reference="$escapetool.xml($services.model.serialize($attachmentReference, 'default'))"
-          title="$escapetool.xml($services.localization.render('core.viewers.attachments.download'))"
-        >$escapetool.xml($attachment.filename)</a>
-      </span>
-      <div class="btn-group xwikibuttonlinks">
-        #if ($services.officemanager.isConnected() &&
-            $services.officeviewer.isMimeTypeSupported($attachment.mimeType.toLowerCase()))
-          ## Link to preview office document.
-          <a class="viewlink btn btn-xs btn-default" title="$escapetool.xml($services.localization.render(
-            'core.viewers.attachments.officeView.title'))" target="_blank"
-            href="$attachment.document.getURL('view', $escapetool.url({
-              'xpage': 'office',
-              'attachment': $attachment.filename
-            }))">$services.icon.renderHTML('eye')</a>
```

Comment:
It seems we lost the office document preview. You need to add it to the Actions column. This shows that we will probably have to update the extensions that add attachment actions. Would be nice if we could reduce the changes needed on the extension side, or remove the need for an update if possible. Maybe an UI extension point makes sense also.